src/components/TodoItem.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FaTrash } from 'react-icons/fa';
+import styles from './TodoItem.module.scss';
+
+class TodoItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: false,
+      bufferTitle: props.todo.title,
+    };
+  }
+
+  handleEditing = () => {
+    this.setState({
+      editing: true,
+    });
+  }
+
+  changeUpdate = (e) => {
+    this.setState({
+      bufferTitle: e.target.value,
+    });
+  }
+
+  handleUpdateDone = (event) => {
+    const { setUpdate, todo: { id } } = this.props;
+    const { bufferTitle } = this.state;
+    if (event.key === 'Enter') {
+      setUpdate(bufferTitle, id);
+      this.setState({
+        editing: false,
+      });
+    }
+    if (event.key === 'Escape') {
+      this.setState({
+        editing: false,
+      });
+    }
+  }
```

Comment:
Please, remove the changes to the "buffer title" after the user has chosen to Escape the Edition Mode.

Reply:
Oh, totally. I also like that behavior better, just didn't think about it the time around 😅